### PR TITLE
Renting

### PR DESCRIPTION
**Pull request by Mingling94**
Still in progress, but the header capsule has information and the file list should be populated

### CODE REVIEW

File: app/js/daemonAPI.js [hunk 1]
```diff
@@ -36,7 +36,11 @@ function sendCall(url, type, args, callback) {
 		url: url,
 		type: type,
 		success: function(responseData, textStatus, jqXHR) {
-			callback(null, JSON.parse(responseData), textStatus, jqXHR);
+			try {
+				callback(null, JSON.parse(responseData), textStatus, jqXHR);
+			} catch(e) {
+				console.error(e, 'Ajax response was: ' + responseData);
```
Comment (Mingling94) [inline app/js/daemonAPI.js:44]: Catches improperly constructed JSONs that JSON.parse would normally return a weird error on

File: app/js/daemonManager.js [hunk 2]
```diff
@@ -63,21 +64,21 @@ function DaemonManager() {
...
-			UI.notify('siad is not running!', 'stop');
-			return;
-		}
 		apiCall("/daemon/updates/check", function(err, update) {
 			if (err) {
-				UI.notify('Update check failed!', 'error');
-				return;
+				self.Running = false;
+				// Check again later
+				setTimeout(updatePrompt, 1000);
```
Comment (Mingling94) [inline app/js/daemonManager.js]: Makes it so update check will delay API calls until successful. Won't return "Update check failed!" and will wait the duration that it takes to load up the blockchain. That's desired behavior, right?

File: app/js/pluginManager.js [hunk 3]
```diff
@@ -127,6 +127,9 @@ function PluginManager() {
 					event.args[1].left += $('#sidebar').width();
 					UI.tooltip.apply(null, event.args);
 					break;
+				case 'dialog':
+					IPC.sendToHost('dialog', event.args)
+					break;
```
Comment (Mingling94) [inline app/js/pluginManager.js:141]: Needed for Files plugin, still working on it

File: app/plugins/Files/js/buttons.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	updateFileList(lastLoadedFiles);
+});
+eID('search').onclick = function(e) {
+	eSearch.focus();
+	updateFileList(lastLoadedFiles);
+});
+eID('add-ascii').onclick = function(e) {
+	ui.switchView("add-ascii");
+});
+
```
Comment (Mingling94) [inline app/plugins/Files/js/buttons.js]: This whole file was ported from the old UI but I need to de-jquery it

File: app/js/daemonAPI.js (hunk 1 above)
Comment (DavidVorick) [inline app/js/daemonAPI.js:44]: these comments should be in the code

File: package.json [hunk 5]
```diff
@@ -21,10 +21,10 @@
...
     "lint": "jshint . --verbose --exclude-path=.jshintignore",
     "doc": "jsdoc -c .jsdocrc",
-    "sia-repo": "go get -u github.com/NebulousLabs/Sia/...",
-    "sia-import": "test -d app/dependencies/Sia || mkdir -p app/dependencies/Sia && cp $GOPATH/bin/siad app/dependencies/Sia",
-    "sia-kill": "killall siad; while pgrep siad; do sleep .5; done;",
-    "sia-reimport": "(cd $GOPATH/src/github.com/NebulousLabs/Sia && make) && npm run sia-kill && npm run sia-import",
+    "s-repo": "go get -u github.com/NebulousLabs/Sia/...",
+    "s-import": "test -d app/dependencies/Sia || mkdir -p app/dependencies/Sia && cp $GOPATH/bin/siad app/dependencies/Sia",
+    "s-kill": "killall siad; while pgrep siad; do sleep .5; done;",
+    "s-reimport": "(cd $GOPATH/src/github.com/NebulousLabs/Sia && make) && npm run sia-kill && npm run sia-import",
```
Comment (DavidVorick) [inline package.json]: probably best if you undo this change. Seems like a shortcut to you, but would be confusing to newbies.

**Comment by lukechampine** [comment]
Is this ready to go?

**Comment by Mingling94** [comment]
@lukechampine No it isn't. I've been getting caught up on various things

**Comment by Mingling94** [comment]
@lukechampine Now it's ready

File: app/plugins/Files/js/buttons.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+// Used to hide subsequent steps when selecting and earlier one
+function hideSteps(number) {
+	var c = eID('step' + number).children;
+	for (var i = 0; i < c.length; i++) {
+		if(!hidden(c[i])) {
+			hide(c[i]);
+		}
+	}
+}
```
Comment (lukechampine) [inline app/plugins/Files/js/buttons.js:11]: just curious; could you use `map` here? i.e. something like:

``` js
function hideSteps(number) {
    eID('step' + number).children.map(hide);
}
```

File: app/plugins/Files/js/lifecycle.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+	}
+}
+
+// Regularly update the file library and status
+function update() {
+	IPC.sendToHost('api-call', '/renter/files/list', 'update-list');
+	IPC.sendToHost('api-call', '/renter/status', 'update-status');
+	
+	updating = setTimeout(update, 15000);
+}
```
Comment (lukechampine) [inline app/plugins/Files/js/lifecycle.js:72]: As I understand it, this calls `update` recursively after a timeout? Might you want to use `setInterval` instead?

File: app/plugins/Files/js/loaders.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+	if (!savePath) {
+		return;
+	}
+	notify('Downloading ' + nickname + ' to '+ savePath +' folder', 'download');
+	IPC.sendToHost('api-call', {
+		url: '/renter/files/download',
+		type: 'GET',
+		args: {
+			nickname: nickname,
+			destination: savePath
```
Comment (lukechampine) [inline app/plugins/Files/js/loaders.js:18]: try to be consistent about lines ending in `,`. `go fmt` enforces `,` on every line of a struct, so I'd prefer to use that style in the UI as well. The advantage is that if you reorder some lines, there's no risk of invisibly breaking something.

**Comment by lukechampine** [comment]
Is there a reason why we need a separate `.jshintrc` for each plugin? Do they have plugin-specific settings?

**Comment by DavidVorick** [comment]
merge conflict

**Comment by DavidVorick** [comment]
lol that's the dumbest merge conflict I've ever seen. A comment, no code, and it's just a formatting thing not even an actual change.

I fixed it and put it on a different branch. PR looks good, I'm going to merge that branch. Also, the `New File` page is quite spiffy.

Thanks for putting in the time on this, was great working with you this summer.
